sanic/app.py

```diff
@@ -1276,6 +1289,10 @@ async def _run_request_middleware(
     async def _run_response_middleware(
         self, request, response, request_name=None
     ):  # no cov
+        if response.middlewares_executed:
+            return response
+        else:
+            response.middlewares_executed = True
```

Comment:
This seems like a bit of a hack to me. Why not try to prevent them from being called multiple times in the first place?

Reply:
Let me thinking about this

Reply:
Now it is being checked in `request.respond`, where middleware is executed. Do you think it is better now?

Reply:
will look.

Reply:
Thanks, I think I can consider this as resolved. 😊